docs/Macros.md

```diff
@@ -98,4 +98,103 @@ the resulting code. `macro-log` is useful for tracing your macro, a form of
 
 ## Inner Workings
 
-TODO
+The Carp compiler is split in a few different stages. The diagram below
+illustrates the flow of the compiler.
+
+![The compiler passes](./compiler-passes.svg)
+
+The dynamic evaluator is arguably one of the most central pieces of the Carp
+compiler. It orchestrates macro expansion, borrow checking, and type inference,
+as it encounters forms that have requirements for these services, such as
+function definitions, variables, or `let` bindings.
+
+As such, the evaluator serves many purposes. While you might start out trying
```

Comment:
Might be able to pare this down to just "Understanding the evaluator will give you a lot of insight into how Carp works."

I think the rest mostly restates the points of the previous paragraph.

Reply:
I condensed it, thank you for pointing this out!